rust/src/lib.rs

```diff
@@ -14,6 +14,8 @@
 
 pub mod encodings;
 pub mod io;
+pub mod metadata;
```

Comment:
should `metadata` be private mod for now?

Reply:
If we private it, then `metadata_test.rs` can't read it. Do you have recommendations on how to manage the test files for private modules?

Reply:
updated